src/Examples/UnitModels/cstr_testing.ipynb

```diff
@@ -0,0 +1,1123 @@
+    "\n",
+    "![](egprod_flowsheet.png)\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Importing required pyomo and idaes components\n",
+    "\n",
+    "\n",
+    "To construct a flowsheet, we will need several components from the pyomo and idaes package. Let us first import the following components from Pyomo:\n",
+    "- Constraint (to write constraints)\n",
+    "- Var (to declare variables)\n",
+    "- ConcreteModel (to create the concrete model object)\n",
+    "- Expression (to evaluate values as a function of variables defined in the model)\n",
+    "- Objective (to define an objective function for optimization)\n",
+    "- SolverFactory (to solve the problem)\n",
+    "- TransformationFactory (to apply certain transformations)\n",
+    "- Arc (to connect two unit models)\n",
+    "\n",
+    "For further details on these components, please refer to the pyomo documentation: https://pyomo.readthedocs.io/en/latest/\n",
+    "\n",
+    "From idaes, we will be needing the FlowsheetBlock and the following unit models:\n",
+    "- Mixer\n",
+    "- Heater\n",
+    "- CSTR\n",
+    "\n",
+    "We will also be needing some utility tools to put together the flowsheet and calculate the degrees of freedom.\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from pyomo.environ import (Constraint,\n",
+    "                           exp,\n",
+    "                           Var,\n",
+    "                           ConcreteModel,\n",
+    "                           Expression,\n",
+    "                           Objective,\n",
+    "                           SolverFactory,\n",
+    "                           TransformationFactory,\n",
+    "                           value,\n",
+    "                           units as pyunits)\n",
+    "from pyomo.network import Arc, SequentialDecomposition\n",
+    "\n",
+    "from idaes.core import FlowsheetBlock\n",
+    "from idaes.generic_models.properties.core.generic.generic_property import (\n",
+    "        GenericParameterBlock)\n",
+    "from idaes.generic_models.properties.core.generic.generic_reaction import (\n",
+    "        GenericReactionParameterBlock)\n",
+    "from idaes.generic_models.unit_models import (Mixer,\n",
+    "                                              Heater,\n",
+    "                                              CSTR)\n",
+    "\n",
+    "from idaes.core.util.constants import Constants as c\n",
+    "from idaes.core.util import get_solver\n",
+    "from idaes.core.util.model_statistics import degrees_of_freedom\n",
+    "from idaes.core.util.initialization import (propagate_state,\n",
+    "                                            fix_state_vars,\n",
```

Comment:
I did not see `fix_state_vars`, `revert_state_vars` or the logger being used here.